sway-core/src/semantic_analysis/ast_node/declaration/auto_impl/mod.rs

```diff
@@ -45,19 +47,23 @@ where
     /// Parses `input` into the expected [Parse] type.
     /// The resulted [Parse] has source id set to autogenerated source id
     /// within the program represented by the `program_id`.
-    fn parse<T>(engines: &Engines, program_id: Option<ProgramId>, input: &str) -> T
+    fn parse<T>(engines: &Engines, original_source_id: Option<&SourceId>, src: &str) -> T
     where
         T: Parse,
     {
         // Uncomment this to see what is being generated
-        // println!("{}", input);
+        // println!("{}", src);
```

Comment:
nit: it maybe worthwhile putting such useful prints/debugging behind a feature flag - so one does not need to comment out/in for PRs.
Maybe in another PR.